<!-- -->
<launch>
  <arg name='host' default='localhost'/>
  <arg name='port' default='2000'/>
  <arg name='town' default=''/>
  <arg name="vehicle_filter" default='vehicle.toyota.prius*' />
  <arg name="sensor_definition_file" default='$(find carla_autoware_bridge)/config/sensors.json'/>
  <arg name='role_name' default='ego_vehicle'/>
  <!-- use comma separated format "x,y,z,roll,pitch,yaw" -->
  <arg name='spawn_point' default=''/>
  <!-- use comma separated format "x,y,z,quat_x,quat_y,quat_z, quat_w" -->
  <arg name="goal_point" default='' />
  <arg name="map_path" default="$(find carla_autoware_bridge)/autoware-contents/maps/point_cloud_maps"/>

  <!--
    #################
    # CARLA Clients #
    #################
  -->
  <!-- ros-bridge -->
  <include file="$(find carla_ros_bridge)/launch/carla_ros_bridge.launch">
    <arg name='host' value='$(arg host)'/>
    <arg name='port' value='$(arg port)'/>
    <arg name='town' value='$(arg town)'/>
  </include>

  <!-- carla autoware ego vehicle -->
  <include file="$(find carla_ego_vehicle)/launch/carla_ego_vehicle.launch" >
    <arg name='host' value='$(arg host)'/>
    <arg name='port' value='$(arg port)'/>
    <arg name='vehicle_filter' value='$(arg vehicle_filter)'/>
    <arg name="sensor_definition_file" value='$(arg sensor_definition_file)'/>
    <arg name='role_name' value='$(arg role_name)'/>
    <arg name='spawn_point' value='$(arg spawn_point)'/>
  </include>

  <!-- mission planning -->
  <!--
  <include file="$(find carla_waypoint_publisher)/launch/carla_waypoint_publisher.launch" >
    <arg name='host' value='$(arg host)'/>
    <arg name='port' value='$(arg port)'/>
    <arg name="role_name" value='$(arg role_name)'/>
    <arg name="goal_point" value="$(arg goal_point)" />
  </include>
  -->

  <!--
    ##############################
    # Map loading                #
    ##############################
  -->
  <include file="$(find carla_autoware_bridge)/launch/carla_points_map_loader.launch">
      <arg name="maps_path" value="$(arg map_path)"/>
  </include>


  <!--
    ##############################
    # ROS remapping + conversion #
    ##############################
  -->
  <!-- <include file="$(find carla_autoware_bridge)/launch/carla_autoware_bridge_common.launch">
    <arg name="role_name" value='$(arg role_name)'/>
  </include> -->
</launch>
